tutorials/3d/portals/intermediate.rst

```diff
@@ -0,0 +1,80 @@
+
+ROAMING
+^^^^^^^
+Roaming mode is for objects that can move between rooms. Things like players you will want to be roaming. These are more expensive to calculate than ``STATIC`` or ``DYNAMIC`` modes, because the system has to keep track of which room a roaming object is within.
+
+GLOBAL
+^^^^^^
+Global mode is for objects that you don't want occlusion culled at all. Things like a main player's weapon, bullets and particle effects are good candidates for ``GLOBAL`` mode.
+
+IGNORE
+^^^^^^
+Ignore is a special mode for objects that will be essentially free in the system. Manual bounds (``Bound_``) get converted to ignore portal mode automatically. They don't need to show up during the game, but are kept in the scene tree in case you need to convert the level multiple times (e.g. in the Editor). You might also choose to use this for objects that you *only* want to show up in the editor (when RoomManager is inactive).
+
+Should you place objects within rooms (in the scene tree) or not?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+``STATIC`` and ``DYNAMIC`` objects are ideally placed within rooms in the scene tree. The system needs to know which room they are in during conversion as it assumes they will never change room. Placing them within rooms in the scene tree allows you to explicitly tell the system where you want them.
+
+Autoplace
+^^^^^^^^^
+However, for ease of use, it is also possible to place ``STATIC`` and ``DYNAMIC`` objects _outside_ the rooms, but within the roomlist branch. The system will attempt to **autoplace** the objects into the appropriate room. This works in most cases but if in doubt, use the explicit approach, especially when dealing with internal rooms, which have some restrictions for sprawling objects.
+
+.. image:: img/freeform.png
+
+Note that if you place ``STATIC`` and ``DYNAMIC`` objects outside of rooms, they will not contribute to the room bound. So if you are using the room geometry to derive the bound, tables and chairs can be placed outside the room, but walls and floors should be explicitly within the Room's branch of the scene tree, in order to ensure the bound is correct.
+
+``ROAMING`` and ``GLOBAL`` objects you are recommended to maintain in a branch of the scene tree outside of any rooms or the roomlist (they *can* be placed inside the rooms, but to save confusion they are normally better kept on their own branch). There are no restrictions on the placement of ``IGNORE`` objects.
+
+Object Lifetimes
+^^^^^^^^^^^^^^^^
+It is important to note that the lifetime of ``STATIC`` and ``DYNAMIC`` objects is tied to the lifetime of the level, between when you call ``rooms_convert`` to activate the portal system, and calling ``rooms_clear`` to unload the system. This is because quite a bit of pre-processing goes on during the conversion phase in order to render them efficiently.
```

Comment:
```suggestion
Object lifetimes
^^^^^^^^^^^^^^^^

It is important to note that the lifetime of ``STATIC`` and ``DYNAMIC`` objects is tied to the lifetime of the level, between when you call ``rooms_convert()`` to activate the portal system, and calling ``rooms_clear()`` to unload the system. This is because quite a bit of preprocessing goes on during the conversion phase to render them efficiently.
```